src/EditorFeatures/Text/Extensions.TextBufferMapper.cs

```diff
@@ -16,36 +14,23 @@ public static partial class Extensions
         /// </summary>
         private static class TextBufferMapper
         {
-            private static readonly ConditionalWeakTable<ITextSnapshot, ITextSnapshot> s_editorToRoslynSnapshotMap = new ConditionalWeakTable<ITextSnapshot, ITextSnapshot>();
-            private static readonly ConditionalWeakTable<ITextSnapshot, WeakReference<ITextSnapshot>> s_roslynToEditorSnapshotMap = new ConditionalWeakTable<ITextSnapshot, WeakReference<ITextSnapshot>>();
-            private static readonly ConditionalWeakTable<ITextSnapshot, ITextSnapshot>.CreateValueCallback s_createSnapshotCallback = CreateSnapshot;
+            private static readonly ConditionalWeakTable<ITextSnapshot, ITextImage> s_editorToRoslynSnapshotMap = new ConditionalWeakTable<ITextSnapshot, ITextImage>();
+            private static readonly ConditionalWeakTable<ITextImage, WeakReference<ITextSnapshot>> s_roslynToEditorSnapshotMap = new ConditionalWeakTable<ITextImage, WeakReference<ITextSnapshot>>();
+            private static readonly ConditionalWeakTable<ITextSnapshot, ITextImage>.CreateValueCallback s_createSnapshotCallback = CreateSnapshot;
 
-            public static ITextSnapshot ToRoslyn(ITextSnapshot editorSnapshot)
+            public static ITextImage ToRoslyn(ITextSnapshot editorSnapshot)
             {
                 Contract.ThrowIfNull(editorSnapshot);
 
                 var roslynSnapshot = s_editorToRoslynSnapshotMap.GetValue(editorSnapshot, s_createSnapshotCallback);
-                if (roslynSnapshot == null)
```

Comment:
Can't ever fail.  We no longer need a TextBufferFactory.  We can always get the ITextImage from an ITextSnapshot